x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java

```diff
@@ -87,27 +86,27 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.concurrent.Executor;
-import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
 /**
- * {@link AbstractLookupService} performs a single valued {@code LEFT JOIN} for a
- * given input page against another index. This is quite similar to a nested loop
- * join. It is restricted to indices with only a single shard.
+ * {@link AbstractLookupService} performs a {@code LEFT JOIN} for a given input
+ * page against another index that <strong>must</strong> have only a single
```

Comment:
Is this correct? For enrich indices, this assumption is wrong, isn't it? And this abstract service class is also a super class for the enrich lookup service, no?

Reply:
Enrich indices have a single shard. They are funny though, because the *source* index of the enrich index can be whatever. But when you run the enrich policy we copy everything into a single-shard index and *that* has to be a single-shard.